src/lib.rs

```diff
@@ -421,40 +439,45 @@ enum Repr {
 }
 
 impl Repr {
-    fn new<T>(text: T) -> Self
+    /// This function tries to create a new Repr::Inline or Repr::Substring
+    /// If it isn't possible, this function returns None
+    fn new_on_stack<T>(text: T) -> Option<Self>
```

Comment:
I think we can fold this back into `new`, as it was before (`Repr::new_on_stack(s.as_ref()).unwrap_or_else(|| Repr::Heap(s))` is used in two places).

Reply:
The `unwrap_or_else` closure differs in the two, since one allocates an arc, the other reuses it

Reply:
Oh, right, sorry.